nullaway/src/main/java/com/uber/nullaway/handlers/contract/ContractCheckHandler.java

```diff
@@ -0,0 +1,150 @@
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.uber.nullaway.handlers.contract;
+
+import static com.uber.nullaway.handlers.contract.ContractUtils.getContractFromAnnotation;
+import static com.uber.nullaway.handlers.contract.ContractUtils.reportMatch;
+
+import com.google.common.base.Preconditions;
+import com.google.errorprone.VisitorState;
+import com.google.errorprone.util.ASTHelpers;
+import com.sun.source.tree.MethodTree;
+import com.sun.source.tree.ReturnTree;
+import com.sun.source.util.TreePath;
+import com.sun.tools.javac.code.Symbol;
+import com.uber.nullaway.NullAway;
+import com.uber.nullaway.NullabilityUtil;
+import com.uber.nullaway.Nullness;
+import com.uber.nullaway.handlers.BaseNoOpHandler;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * This Handler parses the jetbrains @Contract annotation and tries to check if the contract is
+ * followed.
+ *
+ * <p>Currently, it supports the case when there is only one clause in the contract. The clause of
+ * the form in which all the elements of the antecedent are either of "_", "null" or "!null", and
+ * the consequent is "!null" is supported. The handler checks and warns under the conditions of the
+ * antecedent if the consequent is "!null" and there is a return statement with "nullable" or "null"
+ * expression.
+ */
+public class ContractCheckHandler extends BaseNoOpHandler {
+
+  private Set<TreePath> supportedMethods = new HashSet<>();
+
+  @Override
+  public void onMatchMethod(
+      NullAway analysis, MethodTree tree, VisitorState state, Symbol.MethodSymbol methodSymbol) {
+    Symbol.MethodSymbol callee = ASTHelpers.getSymbol(tree);
+    Preconditions.checkNotNull(callee);
+    // Check to see if this method has an @Contract annotation
+    String contractString = getContractFromAnnotation(callee);
+
+    if (contractString != null) {
+      // Found a contract, lets parse it.
+      String[] clauses = contractString.split(";");
```

Comment:
Should we have a common utility method for parsing contract annotations rather than having multiple places where we parse the string?

Reply:
Done!